Sources/ActorSingletonPlugin/ActorSingletonPlugin.swift

```diff
@@ -113,18 +136,39 @@ public struct ActorSingletonControl {
         return singletonPlugin
     }
 
-    /// Defines a singleton `behavior` and indicates that it can be hosted on this node.
-    public func host<Message>(_ type: Message.Type, name: String, props: _Props = _Props(), _ behavior: _Behavior<Message>) throws -> _ActorRef<Message> {
-        try self.singletonPlugin.ref(of: type, name: name, system: self.system, props: props, behavior)
+    /// Defines a singleton and indicates that it can be hosted on this node.
+    public func proxy<Act>(
+        _ type: Act.Type,
+        name: String,
+        props: _Props = _Props(),
+        _ factory: @escaping (ClusterSystem) async throws -> Act
```

Comment:
The "proxy" method means "always proxy but never host" so it would not have the factory function I think.

Reply:
We can revisit an discuss naming though

Reply:
> "always proxy but never host"

Right. I can't recall why we have this mode though in the original impl. Does it make sense to have it?